src/core/ucc_context.c

```diff
@@ -475,6 +485,9 @@ ucc_status_t ucc_context_create(ucc_lib_h lib,
     b_params.estimated_num_ppn = config->estimated_num_ppn;
     b_params.prefix            = lib->full_prefix;
     b_params.thread_mode       = lib->attr.thread_mode;
+    if (params->mask & UCC_CONTEXT_PARAM_FIELD_OOB) {
+        ctx->rank = params->oob.oob_ep;
```

Comment:
is ctx->rank re-calculated and set again later (at line:579) ?

Reply:
good catch. i've added the "assert" but forgot to remove the line 579.